src/carcockpit/application.hpp

```diff
@@ -0,0 +1,58 @@
+/*
+carcockpit - Car cockpit example GUI project
+
+Copyright (C) 2024 Gagistech Oy <gagisechoy@gmail.com>
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+/* ================ LICENSE END ================ */
+
+#pragma once
+
+#include <ruisapp/application.hpp>
+
+#include "shaders/shader_adv.hpp"
+#include "shaders/shader_phong.hpp"
+#include "shaders/shader_skybox.hpp"
+
+using namespace std::string_view_literals;
+
+namespace carcockpit {
+
+class application : public ruisapp::application
+{
+public:
+	application(bool window, std::string_view res_path);
+
+	void toggle_camera();
+	void toggle_normal_mapping();
+
+	static constexpr std::string_view app_name = "carcockpit"sv;
+
+	static application& inst()
+	{
+		return dynamic_cast<application&>(ruisapp::application::inst());
```

Comment:
I think this should work with `static_cast`. Try replacing `dynamic_cast` with `static_cast`. If linter complains, suppress it.

Reply:
changed to static_cast and suppressed